### PR TITLE
Player input

### PR DESCRIPTION
**Pull request by ynks**
In order to merge this, two things need to be implemented

- [x] #11 
- [x] #12

### CODE REVIEW

**Comment by ynks** [comment]
@Omagoge it's your time to implement #12

File: Assets/Player/MouseAim.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class MouseAim : MonoBehaviour
+{
+    PlayerIA _Input;
```
Comment (ynks) [inline Assets/Player/MouseAim.cs:8]: Rename variables using _camelCase

File: Assets/Player/PlayerInput.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    
+    [Header("Player Movement")]
+    [SerializeField] private float currentSpeed;
+    [SerializeField] private MotionController movementController;
+    [SerializeField] private float inputSmoothTime = 0.2f;
+    private Vector2 _inputDir;
+    private Vector2 _directionTarget;
+    private Vector2 _directionSmooth;
+    private Vector2 _velocity;
+
```
Comment (ynks) [inline Assets/Player/PlayerInput.cs:17]: Implement rotation here

File: Assets/Player/MouseAim.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        _Input.Disable();
+    }
+    private void OnAim (InputAction.CallbackContext context)
+    {
+        //Gets the mouse position in comparison with the camera
+        _MousePos = _Camera.ScreenToWorldPoint(context.ReadValue<Vector2>());
+    }
+
+    
+    private void FixedUpdate()
```
Comment (ynks) [inline Assets/Player/MouseAim.cs:40]: Do this in update better

File: Assets/Player/MouseAim.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        //Updates the vector of the direction the player is facing
+        Vector2 playerDirection = _MousePos - _Rigidbody.position;
+        //Calculates the angle needed for that movement and turns the player into the new direction
+        float angle = Mathf.Atan2 (playerDirection.y, playerDirection.x) * Mathf.Rad2Deg;
+        _Rigidbody.MoveRotation(angle);
```
Comment (ynks) [inline Assets/Player/MouseAim.cs:46]: Use linear interpolation to smooth movement

**Review by ynks — COMMENTED** [review]
@Omagoge see comments for what to change, rotation needs to be added linear interpolation

**Review by dariormt — APPROVED** [review]
👌

**Pull request merged**